bin/pdagentd.py

```diff
@@ -404,6 +404,6 @@ def make_rotating_handler(fname, maxBytes, backupCount, level):
 
 
 # ---- Daemonize and run agent
-
-daemonize(pidfile, umask=_DEFAULT_UMASK)
+if not (len(sys.argv) > 1 and sys.argv[1] == '-f'):
+    daemonize(pidfile, umask=_DEFAULT_UMASK)
```

Comment:
Reviewing https://github.com/PagerDuty/pdagent/blob/master/pdagent/thirdparty/daemon.py  I'm guessing that you also want the `pidfile` related logic from there.

Since that is third party code with it's own license, I recommend that you clone daemon.py in the thirdparty directory and modify it as needed.

The else block become something like:

```
else:
    foo(pidfile, _DEFAULT_UMASK)
 ```

Naming is left as an exercise for the reader. :trollface: